src/lib/NonPartitionedSessionStorage.js

```diff
@@ -0,0 +1,190 @@
+/* global Nimiq */
+/* global CookieJar */
+/* global CookieStorage */
+/* global BrowserDetection */
+
+class NonPartitionedSessionStorage {
+    /**
+     * @param {string} name
+     * @param {Uint8Array} data
+     * @param {Uint8Array} [encryptionKey]
+     * @returns {Promise<Uint8Array | null>} New CookieStorage encryption key, if one was created.
+     */
+    static async set(name, data, encryptionKey) {
+        const isSessionStoragePartitioned = await this.isSessionStoragePartitioned();
+        // Delete old data entries that might not have been migrated over yet.
+        this.delete(name);
+
+        if (isSessionStoragePartitioned) {
+            // Use CookieStorage. We currently only support encrypted CookieStorage entries, to encourage encryption.
+            return CookieStorage.set(name, data, {
+                encryptionKey,
+                maxAge: NonPartitionedSessionStorage.COOKIE_MAX_AGE,
+                namespace: CookieJar.Cookie.NAMESPACE_NON_PARTITIONED_SESSION_STORAGE,
+            });
+        }
+        sessionStorage.setItem(
```

Comment:
Extra blank line to show this is the "else" block.
```suggestion
        }
        
        sessionStorage.setItem(
```